addons/purchase/report/purchase_report.py

```diff
@@ -179,16 +179,19 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
             res = [{}]
 
         if avg_days_to_purchase:
-            query = """ SELECT AVG(days_to_purchase.po_days_to_purchase)::decimal(16,2) AS avg_days_to_purchase
-                          FROM (
-                              SELECT extract(epoch from age(po.date_approve,po.create_date))/(24*60*60) AS po_days_to_purchase
-                              FROM purchase_order po
-                              WHERE po.id IN (
-                                  SELECT "purchase_report"."order_id" FROM %s WHERE %s)
-                              ) AS days_to_purchase
+
+            query = """ SELECT AVG(extract(epoch from age(po.date_approve,po.create_date))/(24*60*60))::decimal(16,2) AS avg_days_to_purchase
+                        FROM purchase_order po
+                        WHERE id in (SELECT "purchase_report"."order_id" FROM %s WHERE %s)
                     """
 
-            subdomain = domain + [('company_id', '=', self.env.company.id), ('date_approve', '!=', False)]
+            # Ensure access rules are correctly applied (Partial copy of models.py/_apply_ir_rules)
+            # In standard, only applies multi-company ir_rule.
+            rules_domain = self.env["ir.rule"]._compute_domain(self._name, 'read') if not self.env.su else []
+            subdomain = AND([
+                domain,
+                AND([rules_domain, [('date_approve', '!=', False)]]),
+            ])
```

Comment:
I'm not a SQL or access rules expert so I think this is OK? If it is approved by someone more knowledgeable than I then these 2 changes should probably also be applied to `stock_account_enterprise > stock_report.py:read_group()` since that is what I originally based this code on.